### PR TITLE
Support for temporary span-like classes. Fixes #70.

### LINKED ISSUE #70: Sort span-like classes
To sort an iterable, one currently needs to pass a pair of iterators or an lvalue of a container type. However, there are more and more view and span classes around in popular libraries (GSL, Library Fundamentals TS, Ranges v3), and **cpp-sort** can't handle them in the following scenario:

    int arr[] = { /* ... */ };
    cppsort::sort(gsl::span<int>(arr + 5, arr + 10));

This example is a bit artificial, but it highlights the problem: one might want to create a span on-the-fly and feed it to a sorting function. While we would expect it should be legal and devoid of surprises, it currently does not compile because **cpp-sort** only handles container lvalues and rejects temporaries (which is normal considering it was designed to sort standard library containers).

Just like Ranges v3 and the latest iteration of the Ranges TS, we should be able to take spans and sort the corresponding iterable. That said, it will probably introduce new classes of error harder to spot since it will probably accept temporary standard library containers and try to sort them.

Now, the usual TODO list:
- [x] Add `utility::begin` and `utility::end` functions to handle rvalues.
- [x] Update `utility::size` to handle rvalues.
- [x] Update `sort` and `stable_sort`.
- [x] Update the range overloads of `merge_sorter` and `quick_sorter`.
- [x] Update `sorter_facade`.
- [x] Update fixed-size sorters if needed.
- [x] Update sorter adapters if needed.
- [x] Complete the `bubble_sorter` example.
- [x] Update the documentation to explain that temporary span-like classes are also supported by the library.
- [x] Document `utility::begin` and `utility::end`.

### PR DESCRIPTION
**Pull request by Morwenn**
This commit enhances sorters so that they can also sort temporary ranges. While this is useless at best and error-prone at worst for standard containers, it is required to handle temporary span-like classes such as `gsl::span` that are more and more common in the C++ realm, and that could even be standardized in a few years.

This also introduces `utility::begin` and `utility::end` (directly taken from Eric Niebler's Range v3) to bypass the fact that `std::begin` and `std::end` can't return mutable iterators when given rvalues.

### CODE REVIEW

File: include/cpp-sort/utility/begin_end.h [hunk 1]
```diff
@@ -0,0 +1,347 @@
...
+    using adl_begin_end_detail::rend_fn;
+    using adl_begin_end_detail::crbegin_fn;
+    using adl_begin_end_detail::crend_fn;
+
+    namespace adl_begin_end_detail
+    {
+        using std::begin;
+        using std::end;
+
+        // A reference-wrapped Range is an Range
```
Comment (vendethiel) [inline include/cpp-sort/utility/begin_end.h]: a Range

Comment (Morwenn) [inline include/cpp-sort/utility/begin_end.h]: Wow, tell that to Eric Niebler. That part is 95% stolen :p

Comment (vendethiel) [inline include/cpp-sort/utility/begin_end.h]: Thief!

Comment (Morwenn) [inline include/cpp-sort/utility/begin_end.h]: But an honest one :)

**Pull request merged**